### PR TITLE
Types/FormItem

### LINKED ISSUE #13555: Add TypeScript types to FormItem

### PR DESCRIPTION
**Pull request by k-rajat19**
Closes #13555

#### Changelog

**New**

- add typescript types to the FormItem component

### CODE REVIEW

File: packages/react/src/components/FormItem/FormItem.tsx [hunk 1]
```diff
@@ -10,7 +10,19 @@ import React from 'react';
...
+type FormItemProps = {
+  /**
+   * Provide content to be rendered in the form item
+   */
+  children: React.ReactNode;
+
+  /**
+   * Provide a custom className to be applied to the containing node
+   */
+  className: string;
```
Comment (guidari) [inline packages/react/src/components/FormItem/FormItem.tsx]:
```suggestion
  className?: string;
```

File: packages/react/src/components/FormItem/FormItem.tsx [hunk 2]
```diff
@@ -10,7 +10,19 @@ import React from 'react';
 import classnames from 'classnames';
 import { usePrefix } from '../../internal/usePrefix';
 
-function FormItem({ className, children, ...rest }) {
+type FormItemProps = {
+  /**
+   * Provide content to be rendered in the form item
+   */
+  children: React.ReactNode;
```
Comment (guidari) [inline packages/react/src/components/FormItem/FormItem.tsx]:
```suggestion
  children?: React.ReactNode;
```

**Review by guidari — CHANGES_REQUESTED** [review]
Hey! Thanks for the contribution! 🚀

Maybe we should leave as optional the props, since in the propTypes is not required

**Review by guidari — APPROVED** [review]
LGTM!

**Review by tw15egan — APPROVED** [review]

**Pull request merged**